python/tk_multi_loader/dialog.py

```diff
@@ -1643,40 +1657,52 @@ def _switch_profile_tab(self, new_index, track_in_history):
             self._setup_details_panel([])
 
             # tell the publish view to change
-            self._load_publishes_for_entity_item(selected_item)
+            self._load_publishes_for_entity_items([selected_item])
 
     def _on_treeview_item_selected(self):
         """
         Slot triggered when someone changes the selection in a treeview.
         """
+        selected_items = self._get_selected_entities()
 
-        selected_item = self._get_selected_entity()
+        model = self._entity_presets[self._current_entity_preset].model
 
-        # update breadcrumbs
-        self._populate_entity_breadcrumbs(selected_item)
+        # If nothing is selected, refresh the view.
+        if len(selected_items) == 0:
+            if isinstance(model, SgEntityModel):
+                model.async_refresh()
```

Comment:
I could be wrong it's been a while since I looked at this, but if you don't refresh the view when nothing is selected, then you'll be left with a filtered list of published_files, where you actually want the list to reset to fully unfiltered.